gcloud-java-pubsub/src/main/java/com/google/gcloud/pubsub/spi/SubscriberApi.java

```diff
@@ -0,0 +1,663 @@
+  }
+
+
+  // =============
+  // Service Calls
+  // =============
+
+  // ----- createSubscription -----
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Creates a subscription to a given topic for a given subscriber.
+   * If the subscription already exists, returns ALREADY_EXISTS.
+   * If the corresponding topic doesn't exist, returns NOT_FOUND.
+   *
+   * If the name is not provided in the request, the server will assign a random
+   * name for this subscription on the same project as the topic.
+   *
+   * @param name The name of the subscription. It must have the format
+   * `"projects/{project}/subscriptions/{subscription}"`. `{subscription}` must
+   * start with a letter, and contain only letters (`[A-Za-z]`), numbers
+   * (`[0-9]`), dashes (`-`), underscores (`_`), periods (`.`), tildes (`~`),
+   * plus (`+`) or percent signs (`%`). It must be between 3 and 255 characters
+   * in length, and it must not start with `"goog"`.
+   * @param topic The name of the topic from which this subscription is receiving messages.
+   * The value of this field will be `_deleted-topic_` if the topic has been
+   * deleted.
+   * @param pushConfig If push delivery is used with this subscription, this field is
+   * used to configure it. An empty pushConfig signifies that the subscriber
+   * will pull and ack messages using API methods.
+   * @param ackDeadlineSeconds This value is the maximum time after a subscriber receives a message
+   * before the subscriber should acknowledge the message. After message
+   * delivery but before the ack deadline expires and before the message is
+   * acknowledged, it is an outstanding message and will not be delivered
+   * again during that time (on a best-effort basis).
+   *
+   * For pull delivery this value is used as the initial value for the ack
+   * deadline. To override this value for a given message, call
+   * ModifyAckDeadline with the corresponding ack_id.
+   *
+   * For push delivery, this value is also used to set the request timeout for
+   * the call to the push endpoint.
+   *
+   * If the subscriber never acknowledges the message, the Pub/Sub
+   * system will eventually redeliver the message.
+   *
+   * If this parameter is not set, the default value of 10 seconds is used.
+   */
+  public Subscription createSubscription(String name, String topic, PushConfig pushConfig, int ackDeadlineSeconds) {
+    Subscription request =
+        Subscription.newBuilder()
+        .setName(name)
+        .setTopic(topic)
+        .setPushConfig(pushConfig)
+        .setAckDeadlineSeconds(ackDeadlineSeconds)
+        .build();
+
+    return createSubscription(request);
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Creates a subscription to a given topic for a given subscriber.
+   * If the subscription already exists, returns ALREADY_EXISTS.
```

Comment:
Issue https://github.com/GoogleCloudPlatform/gcloud-java/issues/385 covers this.